### PR TITLE
HPCC-12776 Document .trim=0 option

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia jamesdefabia@lexisnexis.com
@afishbeck Please review

### CODE REVIEW

File: docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml [hunk 1]
```diff
@@ -373,5 +372,36 @@ to
 
       <para></para>
     </sect2>
+
+    <sect2>
+      <title>Roxie Results </title>
+
+      <para>By default, Roxie results from HTTP queries do return empty
```
Comment (afishbeck) [inline docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml]: should be "do not return"?

File: docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml [hunk 2]
```diff
@@ -373,5 +372,36 @@ to
...
+      <para><emphasis role="bold">WsECL RESTful:</emphasis></para>
+
+      <para>http://ip:port/WsEcl/submit/query/roxie/myquery?param1=abc&amp;.trim=0
+      </para>
+
+      <para><emphasis role="bold">WsECL SOAP:</emphasis></para>
+
+      <para>http://ip:port/WsEcl/soap/query/roxie/myquery?.trim=0 </para>
+
+      <para><emphasis role="bold">Roxie Direct SOAP or JSON:</emphasis></para>
```
Comment (afishbeck) [inline docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml]: And roxie direct REST

**Comment by JamesDeFabia** [comment]
@afishbeck Edits made, please have a second look

File: docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml [hunk 3]
```diff
@@ -373,5 +372,50 @@ to
...
+
+      <para><programlisting>http://ip:port/WsEcl/json/query/roxie/myquery?.trim=0</programlisting></para>
+
+      <para><emphasis role="bold">Roxie Direct SOAP or JSON:</emphasis></para>
+
+      <para><programlisting>http://ip:port/target?.trim=0</programlisting></para>
+
+      <para><emphasis role="bold">Roxie Direct RESTful:</emphasis></para>
+
+      <para><programlisting>http://ip:port/roxie?param1=abc&amp;.trim=0</programlisting></para>
```
Comment (afishbeck) [inline docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml]: needs the queryname

File: docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml [hunk 4]
```diff
@@ -373,5 +372,50 @@ to
...
+
+      <para><emphasis role="bold">Form submission:</emphasis></para>
+
+      <para>For a RESTful form submission, add the .trim=0 decoration to your
+      URL.</para>
```
Comment (afishbeck) [inline docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml]: this is true for either roxie or wsecl.

File: docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml [hunk 5]
```diff
@@ -373,5 +372,50 @@ to
...
+      <para>By default, Roxie results from HTTP queries do not return empty
+      elements (tags). If a field's value is empty, the tag is not returned.
+      However, at times an application may expect all tags to be returned, so
+      you can add a URL decoration to override the default behavior. This
+      causes the empty tags to return with a value of an empty string.</para>
+
+      <para><programlisting>.trim=o</programlisting></para>
+
+      <para>The <emphasis role="bold">.trim=0</emphasis> option can be added
+      to WsECL, SOAP, JSON, or Web form queries.</para>
```
Comment (afishbeck) [inline docs/RoxieReference/RoxieRefMods/directAccessToRoxie.xml]: true for both wsecl and roxie.

Perhaps:
"option can be added to SOAP, JSON, URL, or Web form queries for both WsECL and Roxie"

**Comment by JamesDeFabia** [comment]
@afishbeck Please have another look

**Comment by afishbeck** [comment]
@JamesDeFabia looks good.

**Pull request merged**